src/main/java/com/powsybl/openloadflow/ac/outerloop/tap/TransformerRatioManager.java

```diff
@@ -0,0 +1,140 @@
+/**
+ * Copyright (c) 2024, RTE (http://www.rte-france.com)
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * SPDX-License-Identifier: MPL-2.0
+ */
+package com.powsybl.openloadflow.ac.outerloop.tap;
+
+import com.powsybl.openloadflow.network.LfBranch;
+import com.powsybl.openloadflow.network.LfBus;
+import com.powsybl.openloadflow.network.LfNetwork;
+import com.powsybl.openloadflow.network.PiModel;
+import com.powsybl.openloadflow.network.VoltageControl;
+import org.apache.commons.lang3.tuple.Pair;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.HashMap;
+import java.util.Map;
+
+/**
+ * @author Didier Vidal {@literal <didier.vidal-ext at rte-france.com>}
+ */
+public class TransformerRatioManager {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(TransformerRatioManager.class);
+
+    private final boolean useInitialTapPosition;
+
+    public record SharedControl(double maxR1, double minR1, double initialR1) {
+
+        public SharedControl(PiModel piModel) {
+            this(piModel.getMaxR1(), piModel.getMinR1(), piModel.getR1());
+        }
+
+        public SharedControl(double maxR1, double minR1, double a, double b, int count) {
+            this(maxR1, minR1, (minR1 * a + maxR1 * b) / count);
+        }
+    }
+
+    private final Map<String, Pair<Double, SharedControl>> sharedControlByBranchId = new HashMap<>();
```

Comment:
Instead of using this Pair (not really readable) why not moving the r1 to SharedControl ? Just for using a record ?

Reply:
Because as far as I understand a shared control is shared between the branches of a shared control, so the object is shared too.

Reply:
Yes, right. So maybe we should have another class to embed the SharedControl and its r1 value

Reply:
It was a record initially. Doesn't change much. What we need is a multikey.

Reply:
Indeed I removed it because the good naming was not so easy.